opm/autodiff/NewtonSolver.hpp

```diff
@@ -66,7 +66,7 @@ namespace Opm {
         /// \param[in]      param   parameters controlling nonlinear Newton process
         /// \param[in, out] model   physical simulation model
         explicit NewtonSolver(const SolverParameters& param,
-                              PhysicalModel& model);
+                              std::shared_ptr<PhysicalModel> model);
```

Comment:
More to the point, why is there a `shared_ptr<>` in the interface at all given the inferred semantics as detailed in Herb Sutter's [GOTW 91](http://herbsutter.com/2013/06/05/gotw-91-solution-smart-pointer-parameters/)?